calvin/actorstore/systemactors/net/MQTTDynamicSubscriber.py

```diff
@@ -35,57 +35,68 @@ class MQTTDynamicSubscriber(Actor):
             "auth": { "username": <username "password": <password> },
             "will": { "topic": <topic>, "payload": <payload> },
             "transport": <tcp or websocket>,
+            "clean_session": <true|false>
         }
 
     Input:
         client_id : MQTT client ID
         uri : MQTT broker URI (format: schema://host:port)
+        cmd : command keyword ('subscribe'|'unsubscribe')
         topic : topic to subscribe to
         qos : MQTT qos
     Output:
         message : dictionary {"topic": <topic>, "payload": <payload>, "client_id": <client id>}
     """
 
-    @manage(['settings', 'mqtt_dict'])
+    @manage(['settings', 'mqtt_dict', 'queue'])
     def init(self, settings):
         if not settings:
             settings = {}
         self.settings = settings
         self.mqtt_dict = {}
+        self.queue = []
 
     """
-    Read first available MQTT client message
+    Read all available MQTT clients for messages and store them in a FIFO queue
+    The reader will only read the first message in the queue.
+
+    @note The rest of the messages are expected to be read at the next readings
     """
 
-    @stateguard(lambda self:
+    @stateguard(lambda self: self.queue or
                 any(calvinsys.can_read(mqtt) for mqtt in self.mqtt_dict.itervalues()))
     @condition(action_output=['message'])
     def read_message(self):
-        client_id, mqtt = next((client_id, mqtt)
-                               for (client_id, mqtt) in self.mqtt_dict.iteritems()
-                               if calvinsys.can_read(mqtt))
-        message = calvinsys.read(mqtt)
-        # add client id to the message
-        message["client_id"] = client_id
+        message = ""
+        for (client_id, mqtt) in self.mqtt_dict.iteritems():
+            if (calvinsys.can_read(mqtt)):
```

Comment:
Please remove superfluous parentheses (in all if-statements, not only here)

Reply:
Done